bson/src/main/org/bson/ByteBufNIO.java

```diff
@@ -98,7 +99,7 @@ public ByteBuf put(final byte b) {
 
     @Override
     public ByteBuf flip() {
-        buf.flip();
+        ((Buffer) buf).flip();
```

Comment:
It does.  But without the cast we get NoSuchMethodError at runtime.  That's because the JVM requires an exact full-signature match of the method at runtime.  In Java 9, the signature is

    ByteBuffer flip();   // takes advantage of covariant return types

while in Java 8 it inherits the method from Buffer, which is:

   Buffer flip();

The cast here ensures that it works on both Java 8 and 9, by forcing a binding to the superclass method, whose signature has not changed.